resources/views/layout/metadata.blade.php

```diff
@@ -16,23 +21,33 @@
 <meta name="theme-color" content="hsl({{ $currentHue }}, 10%, 40%)"> {{-- @osu-colour-b1 --}}
 
 <meta charset="utf-8">
-<meta name="description" content="{{ $pageDescription ?? osu_trans('layout.defaults.page_description') }}">
+<meta name="description" content="{{ $opengraph['description'] ?? osu_trans('layout.defaults.page_description') }}">
 <meta name="keywords" content="osu, peppy, ouendan, elite, beat, agents, ds, windows, game, taiko, tatsujin, simulator, sim, xna, ddr, beatmania, osu!, osume">
 <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
 <link rel="search" type="application/opensearchdescription+xml" title="osu! search" href="{{ config('app.url') }}/opensearch.xml">
 
-@if (isset($opengraph))
-    <meta property="og:site_name" content="osu! » {{ page_title() }}">
-    <meta property="og:type" content="website">
+<meta property="og:site_name" content="osu! » {{ page_title() }}">
+<meta property="og:type" content="website">
+
+@if (isset($canonicalUrl))
     <meta property="og:url" content="{{ $canonicalUrl }}">
-    <meta property="og:title" content="{{ $opengraph['title'] }}">
-    <meta property="og:image" content="{{ $opengraph['image'] }}">
+@endif
 
-    @if (isset($pageDescription))
-        <meta property="og:description" content="{{ $pageDescription }}">
+@foreach ($opengraph as $key => $value)
+    @if (present($value))
+        @php
+            if ($key === 'title') {
+                $isHtmlString = $value instanceof HtmlString;
+                $value .= ' · '.page_title();
+                if ($isHtmlString) {
+                    $value = new HtmlString($value);
+                }
+            }
+        @endphp
+        <meta property="og:{{ $key }}" content="{{ $value }}">
```

Comment:
a bit more repetitive but a lot shorter. (also the `page_title()` should be escaped)

```suggestion
        @if ($key === 'title')
            <meta property="og:{{ $key }}" content="{{ $value }} · {{ page_title() }}">
        @else
            <meta property="og:{{ $key }}" content="{{ $value }}">
        @endif
```